core/repos/repo.go

```diff
@@ -31,31 +31,99 @@ import (
 	"github.com/go-git/go-git/v5/plumbing"
 	"github.com/gobwas/glob"
 	"io/ioutil"
+	"net/url"
 	"os"
 	"os/exec"
 	"path"
 	"path/filepath"
+	"regexp"
 	"strings"
 )
 
+type IRepo interface {
```

Comment:
Split this interface into an unexported interface (for use within the repos package) and an exported one.